subsys/net/ip/net_context.c

```diff
@@ -758,17 +758,17 @@ int net_context_bind(struct net_context *context, const struct sockaddr *addr,
 		net_sll_ptr(&context->local)->sll_protocol =
 			ll_addr->sll_protocol;
 		if (IS_ENABLED(CONFIG_NET_L2_IEEE802154) &&
-		    ll_addr->sll_protocol == ETH_P_IEEE802154 &&
-		    ll_addr->sll_halen > 0) {
-			if (ll_addr->sll_halen !=
-			    IEEE802154_SHORT_ADDR_LENGTH) {
+		    ll_addr->sll_protocol == ETH_P_IEEE802154) {
+			if (ll_addr->sll_halen != 0) {
 				return -EINVAL;
 			}
-			net_sll_ptr(&context->local)->sll_addr =
-				ll_addr->sll_addr;
-			net_sll_ptr(&context->local)->sll_halen =
-				ll_addr->sll_halen;
-		} else {
+
+#ifdef CONFIG_NET_L2_IEEE802154
```

Comment:
It's already within `IS_ENABLED(CONFIG_NET_L2_IEEE802154)` block, so this #ifdef seems redundant?

Reply:
Fortunately I could solve this differently now so no need to mention a preprocessor constant at all.

The problem was, that the code wouldn't compile without `#ifdef` as `IS_ENABLED()` will let the compiler see the code inside the block which only works if all code referenced inside this block is also visible to the compiler. This wasn't the case for some struct field in this block which was defined conditionally. Using `IS_ENABLED()` for struct fields wastes memory, though, that the compiler cannot optimize away AFAICS. So while I quite like the `IS_ENABLED()` pattern now that I fully understand its purpose it doesn't seem to work in such cases.